process/smartContract/process.go

```diff
@@ -750,6 +751,27 @@ func (sc *scProcessor) processVMOutput(
 		return nil, nil, process.ErrNilTransaction
 	}
 
+	if isTooMuchGasProvided(vmOutput.GasRemaining, gasProvided) {
+		log.Trace("scProcessor.processVMOutput: too much gas has been provided",
+			"hash", txHash,
+			"nonce", tx.GetNonce(),
+			"value", tx.GetValue(),
+			"sender", tx.GetSndAddr(),
+			"receiver", tx.GetRcvAddr(),
+			"gas limit", tx.GetGasLimit(),
+			"gas price", tx.GetGasPrice(),
+			"gas provided", gasProvided,
+			"gas remained", vmOutput.GasRemaining,
+			"gas used", gasProvided-vmOutput.GasRemaining,
+			"return code", vmOutput.ReturnCode.String(),
+			"return message", vmOutput.ReturnMessage,
+		)
+
+		vmOutput.ReturnMessage = fmt.Sprintf("too much gas has been provided: gas used = %d, gas remained = %d",
```

Comment:
do not rewrite return message - just add to the existing return message

Reply:
done